### DISCUSSION TITLE
Any way to embed CPython?

### DISCUSSION THREAD

**Discussion by retsyo** [pascalabcnet/pascalabcnet#3234] [Q&A]
Since we can make GUI with pascalabcnet quickly and release a small EXE, it is possible to program the GUI in pascalabcnet, but main duty routine in python, well we know there are bundles packages/libs for python in various fields.

The only problem is, how to let pascalabcnet transform data to python and then recieve data from python? No, I don't mean to run python.exe

Thnaks

**Comment by SunSerega** [comment]
Well, logically, there aren't many options here.

You could emulate a Python interpreter from a .Net program.
Maybe someone even wrote an existing C# library for it, which you can easily use from this Pascal? Good luck.

Though at that point, maybe it would be easier to find libraries for your fields made for .Net...
If it's something performance-intensive - the packages you are using are most likely already a Python wrapper of C++ or C# code. And that code would be easier to wrap than figuring out Python-to-.Net interop.

But if you still want to have actual Python (and not emulation) in the equation, you need python.exe.
In that case, look into Pipes. They are like sockets but for processes on the same PC.
Communication using pipes is generally less scuffed than using command line args and in/out/err streams.

P.S. Ah, actually... The main maintainer of this repo is a uni professor, and last year, his students made this:
https://github.com/AlexanderZemlyak/pascalabcnet/tree/SPython
This branch contains a compiler extension that allows compiling Python files and .pas files into the same .exe by generating a common semantics tree.
Basically before starting to generate .exe, representations of e.g. functions from Pascal and Python files are combined into a common form.
I'm sure you can immediately imagine the limitations of that approach. It's obviously not a full Python language. And I honestly don't know enough to tell you if you'd be able to use Python packages with that.
I think there was also a packaged version of this compiler extension, which you could add to an existing install of Pascal (just by dropping .dll-s), but I can't find it now.
